src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/processhearingrecordingrequest/ProcessHearingRecordingRequestAboutToSubmitHandler.java

```diff
@@ -65,54 +60,43 @@ public boolean canHandle(CallbackType callbackType, Callback<SscsCaseData> callb
-            case DWP:
-                DynamicList dwpRequestStatus = processHearingRecordingRequestValue.getDwp();
-                if (dwpRequestStatus != null) {
-                    status = dwpRequestStatus.getValue().getCode();
-                    break;
-                } else {
-                    return;
-                }
-                //TODO:Add unit tests for this, should be part of SSCS-9239
-            case APPELLANT:
-                DynamicList appellantRequestStatus = processHearingRecordingRequestValue.getAppellant();
-                if (appellantRequestStatus != null) {
-                    status = appellantRequestStatus.getValue().getCode();
-                    break;
-                } else {
-                    return;
-                }
-            case JOINT_PARTY:
-                DynamicList jointPartyStatus = processHearingRecordingRequestValue.getJointParty();
-                if (jointPartyStatus != null) {
-                    status = jointPartyStatus.getValue().getCode();
-                    break;
-                } else {
-                    return;
-                }
-            default:
+        if (isPartyItemSet(processHearingRecordingRequestValue.getDwp())) {
+            processHearingRecordingsRequestsForParty(sscsCaseData, PartyItemList.DWP, sscsHearingRecordingCaseData,
+                    processHearingRecordingRequestValue, processHearingRecordingRequestValue.getDwp()
+                            .getValue().getCode());
         }
+        if (isPartyItemSet(processHearingRecordingRequestValue.getAppellant())) {
+            processHearingRecordingsRequestsForParty(sscsCaseData, PartyItemList.APPELLANT, sscsHearingRecordingCaseData,
+                    processHearingRecordingRequestValue, processHearingRecordingRequestValue.getAppellant()
+                            .getValue().getCode());
+        }
+        if (isPartyItemSet(processHearingRecordingRequestValue.getJointParty())) {
+            processHearingRecordingsRequestsForParty(sscsCaseData, PartyItemList.JOINT_PARTY, sscsHearingRecordingCaseData,
+                    processHearingRecordingRequestValue, processHearingRecordingRequestValue.getJointParty()
+                            .getValue().getCode());
+        }
+        if (isPartyItemSet(processHearingRecordingRequestValue.getRep())) {
+            processHearingRecordingsRequestsForParty(sscsCaseData, PartyItemList.REPRESENTATIVE, sscsHearingRecordingCaseData,
+                    processHearingRecordingRequestValue, processHearingRecordingRequestValue.getRep()
+                            .getValue().getCode());
+        }
+    }
 
-        organiseHearingRequestsLists(sscsCaseData, partyItemList, sscsHearingRecordingCaseData,
-                processHearingRecordingRequestValue, status);
+    private boolean isPartyItemSet(DynamicList item) {
+        if (nonNull(item) && nonNull(item.getValue()) && nonNull(item.getValue().getCode())) {
```

Comment:
Can probably remove the if statement here and just use 'return'

Reply:
done